core/matching/orderbook.go

```diff
@@ -747,10 +747,7 @@ func (b *OrderBook) ReplaceOrder(rm, rpl *types.Order) (*types.OrderConfirmation
 	if _, err := b.CancelOrder(rm); err != nil {
 		return nil, err
 	}
-	// Because other collections might be pointing at the original order
-	// use it's memory when inserting the new version
-	*rm = *rpl
-	return b.SubmitOrder(rm)
```

Comment:
Should not be needed anymore as I suppose that was needed for the pegged orders. So would rather avoid using that

Reply:
Probably mixing 3 things up in my mind, but there's something about this deref/assign... Perhaps this is part of some trickery that gives the caller (or some caller higher up the call chain) access to the same memory so we can update market/original prices for example when updating LP order prices?

Probably not, but just thought I'd mention it